src/main/java/org/betonquest/betonquest/compatibility/fakeblock/ShowGroupEvent.java

```diff
@@ -0,0 +1,44 @@
+package org.betonquest.betonquest.compatibility.fakeblock;
+
+import com.briarcraft.fakeblock.api.service.PlayerGroupService;
+import org.betonquest.betonquest.api.profiles.Profile;
+import org.betonquest.betonquest.api.quest.event.Event;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.bukkit.plugin.RegisteredServiceProvider;
+
+import java.util.List;
+
+/**
+ * Shows the FakeBlock groups to the player.
+ */
+public class ShowGroupEvent implements Event {
+    /**
+     * The groupNames that should be shown to the player.
+     */
+    private final List<String> groupNames;
+    /**
+     * PlayerGroupService to change group states for the player.
+     */
+    private final RegisteredServiceProvider<PlayerGroupService> playerGroupService;
+
+    /**
+     * Create the showgroup event.
+     *
+     * @param groupNames         is a list that contains Strings of group names that should be shown to the player.
+     * @param playerGroupService the FakeBlock PlayerGroupService.
+     */
+    public ShowGroupEvent(final List<String> groupNames, final RegisteredServiceProvider<PlayerGroupService> playerGroupService) {
+        this.groupNames = groupNames;
+        this.playerGroupService = playerGroupService;
+    }
+
+    @Override
+    public void execute(final Profile profile) throws QuestRuntimeException {
+        if (groupNames.isEmpty()) {
```

Comment:
I think this is unreachable code.
If `instruction.next()` is null, `instruction.getArray()` will return an empty string. Therefore the `groupNames` list will always contain at least one element.
If this is the case, `checkForNotExistingGroups()` would throw an `InstructionParseException` before this code is reached because the group `""` probably won't exist.

I have removed this and the other check in my commit.